core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -801,20 +800,19 @@ class TransactionsTest extends IntegrationTestHarness {
       producer.send(TestUtils.producerRecordWithExpectedTransactionStatus(testTopic, 0, "4", "4", willBeCommitted = true))
       producer.commitTransaction()
 
-      // Get producerId and epoch after first commit
+      // Second transaction: abort
+      producer.beginTransaction()
+      val successfulFuture = producer.send(TestUtils.producerRecordWithExpectedTransactionStatus(topic1, null, "2", "2", willBeCommitted = false))
+      TestUtils.waitUntilTrue(successfulFuture.isDone, "First message of the second transaction did not complete its send.")
```

Comment:
nit: any reason to use a `waitUntilTrue` here instead of a future.get()`?

Reply:
That's fair. Can make the change.

Reply:
I guess the benefit is that future.get could block for a while -- not sure if there is a global test timeout, but maybe the method itself should have a timeout too?

Reply:
Yeah, I think for the tests it's more important to recover if the tested code gets stuck, than to be optimal from the perf perspective (obviously .get is more efficient because it would just block the thread, vs. waitUntilTrue would just loop and poll).  If we want to fail on send failure (not sure if this is the goal of the test) we can use .get after waitUntilTrue.

Reply:
Oh I missed the continued discussion. I just set the wait on the get to 20 seconds like some of the others in the file. The waitUntilTrue default is 15 seconds so I think this is sufficient.

Reply:
i think the main benefit of the `get()` is that if the send does fail with an exception when you otherwise expect it to succeed, the exception gets thrown and fails the test immediately, rather than continuing with incorrect preconditons.